### PR TITLE
Add new endpoint for keycloak sync

### PR DESCRIPTION
**Pull request by kamil-bielecki-bosch**

### CODE REVIEW

**Review by mmurto — CHANGES_REQUESTED** [review]
As `OSMIDEV-1559` is not publicly available, can you describe the rationale in the commit message bodies? What does removing the Keycloak schema sync from the lifecycle mean for someone maintaining ORT Server instance?

File: core/src/main/kotlin/api/AdminRoute.kt
Comment (mnonnenmacher) [inline core/src/main/kotlin/api/AdminRoute.kt:1]: Commit message:
* Do not use internal ticket ids in public projects.
* Provide a rationale in the body.

File: core/src/main/kotlin/api/AdminRoute.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.slf4j.LoggerFactory
+
+private val logger = LoggerFactory.getLogger(Application::class.java)
+
+fun Route.admin() = route("admin") {
+    route("sync-roles") {
+        val authorizationService by inject<AuthorizationService>()
+
+        get(runPermissionsSync) {
+            requirePermission(Superuser.ROLE_NAME)
```
Comment (mnonnenmacher) [inline core/src/main/kotlin/api/AdminRoute.kt:50]: Use `requireSuperuser()` instead.

File: core/src/main/kotlin/api/AdminRoute.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+            withMdcContext("component" to "core") {
```
Comment (mnonnenmacher) [inline core/src/main/kotlin/api/AdminRoute.kt:52]: This is not required, it is automatically done for all calls by the `CallLogging` plugin configured in `Monitoring.kt`.
Instead, use `withContext(Dispatchers.IO)`.

File: core/src/main/kotlin/plugins/Routing.kt [hunk 3]
```diff
@@ -43,6 +44,7 @@ fun Application.configureRouting() {
                 repositories()
                 runs()
                 versions()
+                admin()
```
Comment (mnonnenmacher) [inline core/src/main/kotlin/plugins/Routing.kt:47]: Put in alphabetic order for overview.

File: core/src/test/kotlin/api/AdminRouteIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                response shouldHaveStatus HttpStatusCode.Accepted
+            }
+        }
+
+        "return Unauthorized if user hve no permissions" {
+            integrationTestApplication {
+                val response = unauthenticatedClient.get("/api/v1/admin/sync-roles")
+                response shouldHaveStatus HttpStatusCode.Unauthorized
+            }
+        }
```
Comment (mnonnenmacher) [inline core/src/test/kotlin/api/AdminRouteIntegrationTest.kt:42]: Use the helper function for testing permissions instead:
```kotlin
        "require the superuser role" {
            requestShouldRequireRole(Superuser.ROLE_NAME) {
                get("/api/v1/admin/sync-roles")
            }
        }
```

File: core/src/main/kotlin/Application.kt
Comment (mnonnenmacher) [inline core/src/main/kotlin/Application.kt:1]: Commit message:

* Do not use internal ticket ids in public projects.
* Provide a rationale in the body.

File: core/src/main/kotlin/Application.kt [hunk 5]
```diff
@@ -21,7 +21,17 @@ package org.eclipse.apoapsis.ortserver.core
...
+import org.eclipse.apoapsis.ortserver.core.plugins.configureDatabase
+import org.eclipse.apoapsis.ortserver.core.plugins.configureHTTP
+import org.eclipse.apoapsis.ortserver.core.plugins.configureKoin
+import org.eclipse.apoapsis.ortserver.core.plugins.configureMetrics
+import org.eclipse.apoapsis.ortserver.core.plugins.configureMonitoring
+import org.eclipse.apoapsis.ortserver.core.plugins.configureOpenApi
+import org.eclipse.apoapsis.ortserver.core.plugins.configureRouting
+import org.eclipse.apoapsis.ortserver.core.plugins.configureSerialization
+import org.eclipse.apoapsis.ortserver.core.plugins.configureStatusPages
+import org.eclipse.apoapsis.ortserver.core.plugins.configureValidation
```
Comment (mnonnenmacher) [inline core/src/main/kotlin/Application.kt:34]: Unrelated change.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/Application.kt (hunk 5 above)
Comment (kamil-bielecki-bosch) [reply core/src/main/kotlin/Application.kt:34]: Moved to another commit

File: core/src/test/kotlin/api/AdminRouteIntegrationTest.kt (hunk 4 above)
Comment (kamil-bielecki-bosch) [reply core/src/test/kotlin/api/AdminRouteIntegrationTest.kt:42]: Nice. Thanks!

**Comment by mnonnenmacher** [comment]
> As `OSMIDEV-1559` is not publicly available, can you describe the rationale in the commit message bodies? What does removing the Keycloak schema sync from the lifecycle mean for someone maintaining ORT Server instance?

@mmurto We decided to drop that part of the PR. There are situations where an immediate role sync on startup makes sense, for example, after replaying a database backup or when deploying a new version with changes to the permission model. Having to manually trigger the sync in such situations would be annoying, especially while testing with Docker Compose.

File: core/src/main/kotlin/api/AdminRoute.kt (hunk 2 above)
Comment (mnonnenmacher) [reply core/src/main/kotlin/api/AdminRoute.kt:51]: Not yet changed.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/api/AdminRoute.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
```
Comment (mnonnenmacher) [inline core/src/main/kotlin/api/AdminRoute.kt:2]: Wrong copyright year.

File: core/src/main/kotlin/apiDocs/AdminDocs.kt [hunk 7]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
```
Comment (mnonnenmacher) [inline core/src/main/kotlin/apiDocs/AdminDocs.kt:2]: Wrong copyright year.

File: core/src/test/kotlin/api/AdminRouteIntegrationTest.kt [hunk 8]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2022 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
```
Comment (mnonnenmacher) [inline core/src/test/kotlin/api/AdminRouteIntegrationTest.kt:2]: Wrong copyright year.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: core/src/test/kotlin/api/AdminRouteIntegrationTest.kt (hunk 8 above)
Comment (kamil-bielecki-bosch) [reply core/src/test/kotlin/api/AdminRouteIntegrationTest.kt]: Done

File: core/src/main/kotlin/apiDocs/AdminDocs.kt (hunk 7 above)
Comment (kamil-bielecki-bosch) [reply core/src/main/kotlin/apiDocs/AdminDocs.kt]: Done

File: core/src/main/kotlin/api/AdminRoute.kt (hunk 6 above)
Comment (kamil-bielecki-bosch) [reply core/src/main/kotlin/api/AdminRoute.kt]: Done

**Review by mnonnenmacher — APPROVED** [review]
